Client/mods/deathmatch/logic/CEvents.h

```diff
@@ -16,10 +16,10 @@
 
 struct SEvent
 {
-    class CLuaMain* pLuaMain;
-    std::string     strName;
-    std::string     strArguments;
-    bool            bAllowRemoteTrigger;
+    std::set<class CLuaMain*> pLuaMainSet;
```

Comment:
I'd argue that using `unordered_set` would be better in terms of memory fragmentation.
Or even better.. `std vector`. Yeah, might sound stupid, but you'd be surprised what the L1 cache can do.

If you go for the `vector` option, you can use `find_if` to check if the event exists, and then insert.
As for the remove `vec.erase(std::remove(vec.begin(), vec.end(), luaMain), vec.end());` will do the trick.

Reply:
I do agree that std::unordered_set makes more sense, I did not know that the default std::set was ordered (I use Qt containers at my job, QSet is already unordered). Just to verify and make sure we are using the most optimal container, I did some performance tests. See the attached image for some of my results, seems like std::unordered_set is quite a lot faster than std::vector (which is also slower than std::set for that matter). Am I missing something here or is the L1 cache thing harder to measure? See also the attached code for my tests, I replicated as close as possible the use-case relevant to this PR.
![unknown](https://user-images.githubusercontent.com/17025996/132054279-2398b1c3-5472-47bd-b757-72fa49abe551.png)
Source code for testing: [https://pastebin.com/j3gYgGYY](https://pastebin.com/j3gYgGYY)

Reply:
Woah, thats some extensive benchmarking!

Your test set is just way too big. Realistically, at most there will be at most 200 - 300 resources running.
And, realistically, not more than 10 resources will try to register the same event.
With this in mind, I think that `std::vector` would be a better choice, for a couple of reasons:
- We dont care about performance here, this function is very rarely used. Unless it takes seconds to execute, its fine.
- We care about memory fingerprint a lot. `unordered_set` has its quirks, and overhead. [MSVC useses double linked lists with a vector](https://stackoverflow.com/a/37177838/15363969), which is just a hell in terms of memory fingerprint. 

In this case, with such a small set of values, a vector will be more efficient in terms of memory, and possibly even performance (because of prefetching).
Obviously, when the number of items is large, a `vector` won't beat a `set`, let alone `unodered_set` in terms of performance, but will always beat them in terms of memory usage.

Reply:
Thank you for the explanation and the link, always nice to learn something! I have pushed the modification to use std::vector instead.

Reply:
Thank you very much. Great!